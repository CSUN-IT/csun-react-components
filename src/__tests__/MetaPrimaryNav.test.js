import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MetaPrimaryNav from '../../src/components/meta/Nav';

configure({ adapter: new Adapter() });

describe('MetaPrimaryNav', () => {
  it('Should be defined', () => {
    expect(MetaPrimaryNav).toBeDefined();
  });

  it('Renders component correctly', () => {
    const tree = shallow(
      <MetaPrimaryNav subBrand={{ title: 'CSUN React Components', to: '' }} navColor={'cream'} />,
    );
    expect(tree).toMatchSnapshot();
  });

  // Commented out since test fails because route is being used on the
  // overall application rather than each individual component.
  //   it('Should verify that it contains 3 child links', () => {
  //     const menuItems = [
  //       {
  //         title: 'Home',
  //         to: '',
  //         exact: true,
  //       },
  //       {
  //         title: 'IT',
  //         to: 'it',
  //       },
  //       {
  //         title: 'Giving',
  //         to: 'giving',
  //       },
  //     ];

  //     const tree = mount(<MetaPrimaryNav menuItems={menuItems} />);
  //     expect(tree.find('.primary-nav__btn')).toHaveLength(3);
  //   });
});
