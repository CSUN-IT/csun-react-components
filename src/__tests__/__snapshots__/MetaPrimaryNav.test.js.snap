// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaPrimaryNav Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaPrimaryNav
    menuItems={Array []}
    navColor="cream"
    subBrand={
      Object {
        "title": "CSUN React Components",
        "to": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="primary-nav__mobile"
        >
          <div
            className="primary-nav__btn"
          >
            <div
              className="bar"
            />
            <div
              className="bar"
            />
            <div
              className="bar"
            />
          </div>
          <a
            className="primary-nav__brand"
            href="https://www.csun.edu"
          >
            <span
              className="sr-only"
            >
              California State University, Northridge (CSUN)
            </span>
          </a>
          <Link
            className="primary-nav__sub-brand"
            replace={false}
            to=""
          >
            CSUN React Components
          </Link>
        </div>
        <ul
          className="primary-nav__links"
        />
      </div>,
      "className": "primary-nav primary-nav--cream ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="primary-nav__mobile"
          >
            <div
              className="primary-nav__btn"
            >
              <div
                className="bar"
              />
              <div
                className="bar"
              />
              <div
                className="bar"
              />
            </div>
            <a
              className="primary-nav__brand"
              href="https://www.csun.edu"
            >
              <span
                className="sr-only"
              >
                California State University, Northridge (CSUN)
              </span>
            </a>
            <Link
              className="primary-nav__sub-brand"
              replace={false}
              to=""
            >
              CSUN React Components
            </Link>
          </div>,
          <ul
            className="primary-nav__links"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="primary-nav__btn"
              >
                <div
                  className="bar"
                />
                <div
                  className="bar"
                />
                <div
                  className="bar"
                />
              </div>,
              <a
                className="primary-nav__brand"
                href="https://www.csun.edu"
              >
                <span
                  className="sr-only"
                >
                  California State University, Northridge (CSUN)
                </span>
              </a>,
              <Link
                className="primary-nav__sub-brand"
                replace={false}
                to=""
              >
                CSUN React Components
              </Link>,
            ],
            "className": "primary-nav__mobile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="bar"
                  />,
                  <div
                    className="bar"
                  />,
                  <div
                    className="bar"
                  />,
                ],
                "className": "primary-nav__btn",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "bar",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "bar",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "bar",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="sr-only"
                >
                  California State University, Northridge (CSUN)
                </span>,
                "className": "primary-nav__brand",
                "href": "https://www.csun.edu",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "California State University, Northridge (CSUN)",
                  "className": "sr-only",
                },
                "ref": null,
                "rendered": "California State University, Northridge (CSUN)",
                "type": "span",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "CSUN React Components",
                "className": "primary-nav__sub-brand",
                "replace": false,
                "to": "",
              },
              "ref": null,
              "rendered": "CSUN React Components",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "primary-nav__links",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="primary-nav__mobile"
          >
            <div
              className="primary-nav__btn"
            >
              <div
                className="bar"
              />
              <div
                className="bar"
              />
              <div
                className="bar"
              />
            </div>
            <a
              className="primary-nav__brand"
              href="https://www.csun.edu"
            >
              <span
                className="sr-only"
              >
                California State University, Northridge (CSUN)
              </span>
            </a>
            <Link
              className="primary-nav__sub-brand"
              replace={false}
              to=""
            >
              CSUN React Components
            </Link>
          </div>
          <ul
            className="primary-nav__links"
          />
        </div>,
        "className": "primary-nav primary-nav--cream ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="primary-nav__mobile"
            >
              <div
                className="primary-nav__btn"
              >
                <div
                  className="bar"
                />
                <div
                  className="bar"
                />
                <div
                  className="bar"
                />
              </div>
              <a
                className="primary-nav__brand"
                href="https://www.csun.edu"
              >
                <span
                  className="sr-only"
                >
                  California State University, Northridge (CSUN)
                </span>
              </a>
              <Link
                className="primary-nav__sub-brand"
                replace={false}
                to=""
              >
                CSUN React Components
              </Link>
            </div>,
            <ul
              className="primary-nav__links"
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="primary-nav__btn"
                >
                  <div
                    className="bar"
                  />
                  <div
                    className="bar"
                  />
                  <div
                    className="bar"
                  />
                </div>,
                <a
                  className="primary-nav__brand"
                  href="https://www.csun.edu"
                >
                  <span
                    className="sr-only"
                  >
                    California State University, Northridge (CSUN)
                  </span>
                </a>,
                <Link
                  className="primary-nav__sub-brand"
                  replace={false}
                  to=""
                >
                  CSUN React Components
                </Link>,
              ],
              "className": "primary-nav__mobile",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="bar"
                    />,
                    <div
                      className="bar"
                    />,
                    <div
                      className="bar"
                    />,
                  ],
                  "className": "primary-nav__btn",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "bar",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "bar",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "bar",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="sr-only"
                  >
                    California State University, Northridge (CSUN)
                  </span>,
                  "className": "primary-nav__brand",
                  "href": "https://www.csun.edu",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "California State University, Northridge (CSUN)",
                    "className": "sr-only",
                  },
                  "ref": null,
                  "rendered": "California State University, Northridge (CSUN)",
                  "type": "span",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "CSUN React Components",
                  "className": "primary-nav__sub-brand",
                  "replace": false,
                  "to": "",
                },
                "ref": null,
                "rendered": "CSUN React Components",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "primary-nav__links",
            },
            "ref": null,
            "rendered": Array [],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
